src/unixfs-data-format.md

```diff
@@ -0,0 +1,652 @@
+whose `Data` field is a UnixFSV1 Protobuf message. For clarity, the specification
+document may represent these nested Protobufs as one object. In this representation,
+it is implied that the `PBNode.Data` field is encoded in a protobuf.
+
+## `dag-pb` Types
+
+A `dag-pb` UnixFS node supports different types, which are defined in
+`decode(PBNode.Data).Type`. Every type is handled differently.
+
+### `dag-pb` `File`
+
+A :dfn[File] is a container over an arbitrary sized amount of bytes. Files are either
+single block or multi-block. A multi-block file is a concatenation of multiple child files.
+
+#### The _sister-lists_ `PBNode.Links` and `decode(PBNode.Data).blocksizes`
+
+The _sister-lists_ are the key point of why IPLD `dag-pb` is important for files. They
+allow us to concatenate smaller files together.
+
+Linked files would be loaded recursively with the same process following a DFS
+(Depth-First-Search) order.
+
+Child nodes must be of type File; either a `dag-pb`:ref[File], or a
+[`raw` block](#raw-blocks).
+
+For example, consider this pseudo-json block:
+
+```json
+{
+  "Links": [{"Hash":"Qmfoo"}, {"Hash":"Qmbar"}],
+  "Data": {
+    "Type": "File",
+    "blocksizes": [20, 30]
+  }
+}
+```
+
+This indicates that this file is the concatenation of the `Qmfoo` and `Qmbar` files.
+
+When reading a file represented with `dag-pb`, the `blocksizes` array gives us the
+size in bytes of the partial file content present in children DAGs. Each index in
+`PBNode.Links` MUST have a corresponding chunk size stored at the same index
+in `decode(PBNode.Data).blocksizes`.
+
+:::warning
+Implementers need to be extra careful to ensure the values in `Data.blocksizes`
+are calculated by following the definition from [`Blocksize`](#decodepbnodedatablocksize).
+:::
+
+This allows for fast indexing into the file. For example, if someone is trying
+to read bytes 25 to 35, we can compute an offset list by summing all previous
+indexes in `blocksizes`, then do a search to find which indexes contain the
+range we are interested in.
+
+In the example above, the offset list would be `[0, 20]`. Thus, we know we only need to download `Qmbar` to get the range we are interested in.
+
+UnixFS parser MUST error if `blocksizes` or `Links` are not of the same length.
+
+#### `decode(PBNode.Data).Data`
+
+An array of bytes that is the file content and is appended before
+the links. This must be taken into account when doing offset calculations; that is,
+the length of `decode(PBNode.Data).Data` defines the value of the zeroth element
+of the offset list when computing offsets.
+
+#### `PBNode.Links[].Name`
+
+This field makes sense only in :ref[Directories] contexts and MUST be absent
+when creating a new file. For historical reasons, implementations parsing
+third-party data SHOULD accept empty values here.
+
+If this field is present and non-empty, the file is invalid and the parser MUST
+error.
```

Comment:
I moved the vague reference to "historical reasons" down here, but I might be hallucinating context; if this suggestion is rejected, I'd recommend otherwise explaining this in enough detail not to bewilder someone with as little context as me or less.

```suggestion
If this field is present and non-empty when DataType is not 1, a conformant parser MUST consider it malformed and error. For historical reasons, the field may be present but empty in the non-directory files of data encoded with earlier versions of IPFS software, so parsers SHOULD accept empty values
to avoid non-interoperability with historical data.
```